lib/src/native/database/connection_pool.dart

```diff
@@ -117,21 +117,15 @@ class SqliteConnectionPool with SqliteQueries implements SqliteConnection {
 
   @override
   Future<T> writeLock<T>(Future<T> Function(SqliteWriteContext tx) callback,
-      {Duration? lockTimeout, String? debugContext}) {
+      {Duration? lockTimeout, String? debugContext}) async {
     if (closed) {
       throw AssertionError('Closed');
     }
     if (_writeConnection?.closed == true) {
       _writeConnection = null;
     }
-    _writeConnection ??= SqliteConnectionImpl(
-        upstreamPort: _upstreamPort,
-        primary: false,
-        updates: updates,
-        debugName: debugName != null ? '$debugName-writer' : null,
-        mutex: mutex,
-        readOnly: false,
-        openFactory: _factory);
+    _writeConnection ??= await _openPrimaryConnection(
```

Comment:
This has the potential for race conditions - if the first queries are multiple concurrent write queries, it could attempt to open multiple write connections.

Reply:
I changed this to keep the initial write connection creation inside `SqliteDatabase` as before.